go/pools/resource_pool.go

```diff
@@ -131,14 +154,16 @@ func (rp *ResourcePool) closeIdleResources() {
 			return
 		}
 
-		if wrapper.resource != nil && idleTimeout > 0 && time.Until(wrapper.timeUsed.Add(idleTimeout)) < 0 {
-			wrapper.resource.Close()
-			wrapper.resource = nil
-			rp.idleClosed.Add(1)
-			rp.active.Add(-1)
-		}
+		func() {
```

Comment:
Was there some reason this needed to be wrapped in a function and not just included inline like it was before?

I find the extra layers of abstraction and the defer of putting the wrapper back into the resource pool to be more confusing this way.

Reply:
Ah that explains it 👍